lib/interface_quda.cpp

```diff
@@ -1898,6 +1898,14 @@ void MatDagMatQuda(void *h_out, void *h_in, QudaInvertParam *inv_param)
   popVerbosity();
 }
 
+namespace quda{
+bool canReuseResidentGauge(QudaInvertParam *param){
+
+  if (gaugePrecise == NULL) errorQuda("Precise gauge field doesn't exist");
```

Comment:
@maddyscientist This line is supposed to check whether a gauge field is allocated and halt if not.